docs/signing-tool-spec.md

```diff
@@ -0,0 +1,129 @@
+# Signing CLI tool
+
+## Background
+
+Code signing is a way to provide tamper resistance to binary files and provide a way of establishing identity. There are different code signing mechanisms, but the most common on Windows and .NET are based on X.509 certificates. Other code signing formats may be based on PGP and other non-certificate-based systems.
+
+There are several technology areas within the Windows and .NET ecosystem that support code signing:
+
+- PE files & certain scripts via Authenticode (dll, exe, ps1, sys)
+- MSIX via Authenticode (msix, msixbundle) & related manifests
+- Visual Studio Extensions (VSIX) via Open Packaging Convention
+- ClickOnce & VSTO via Mage (XML Digital Signatures)
+- NuGet Packages
+
+Today each of these areas has their own tools (SignTool, VISXSignTool, Mage, NuGet) to create the signatures. Each tool has its own set of parameters and are written to assume use of the local Certificate Store API's by default. Without any common implementation, any changes to the code signing landscape requires updates to all of the tools. In May 2022, the CA/B forum updates its baseline requirements to require that all new code signing certificates issued after June 2023 use hardware security modules (HSMs) to prevent key theft. While some HSMs contain CSP/KSP support to expose certificates through the CertStore API's, they frequently contain significant limitations, such as requiring an interactive session to authenticate to the device. This makes signing code in the cloud, on build agents, extremely difficult for mainline scenarios.
+
+There are many HSM cloud services, including Azure Key Vault, that meet the updated key storage requirements, however we do not have first-party support for signing code with those services. There are third-party community solutions to fill this gap: [AzureSignTool](https://github.com/vcsjones/AzureSignTool), [NuGetKeyVaultSignTool](https://github.com/novotnyllc/NuGetKeyVaultSignTool), [OpenOpcSignTool](https://github.com/vcsjones/OpenOpcSignTool), and the .NET Foundation Signing Service (which uses these plus adds additional supported file formats and orchestrates signing the the various types in the right order).
```

Comment:
Since this is a public facing doc, first-party / third-party should be replaced with Microsoft / community / industry / etc.